File: library/src/main/java/com/parasde/library/simpledatepicker/view/CalendarFragmentPagerAdapter.java
package com.parasde.library.simpledatepicker.view;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class CalendarFragmentPagerAdapter extends FragmentStatePagerAdapter {

    private ArrayList<Fragment> fragmentPagerArray = new ArrayList<>();
    private ArrayList<String> fragmentTag = new ArrayList<>();

    CalendarFragmentPagerAdapter(@NonNull FragmentManager fm, int behavior) {
        super(fm, behavior);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        return fragmentPagerArray.get(position);
    }

    @Override
    public int getCount() {
        return fragmentPagerArray.size();
    }

    @Override
    public int getItemPosition(@NotNull Object obj) {
        return POSITION_NONE;
    }

    void addPrevItem(Fragment fragment, String tag) {
        fragmentPagerArray.add(0, fragment);
        fragmentTag.add(0, tag);
        notifyDataSetChanged();
    }

    void addItem(Fragment fragment, String tag) {
        fragmentPagerArray.add(fragment);
        fragmentTag.add(tag);
        notifyDataSetChanged();
    }
}

File: library/src/main/java/com/parasde/library/simpledatepicker/view/CalendarFragmentPagerAdapter.kt
package com.parasde.library.simpledatepicker.view

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class CalendarFragmentPagerAdapter(fm: FragmentManager, behavior: Int = BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) : FragmentStatePagerAdapter(fm, behavior) {
    private var fragmentPagerArray = ArrayList<Fragment>()
    private var fragmentTag = ArrayList<String>()

    override fun getItem(position: Int): Fragment {
        return fragmentPagerArray[position]
    }

    override fun getCount(): Int {
        return fragmentPagerArray.size
    }

    override fun getItemPosition(`object`: Any): Int {
        return POSITION_NONE
    }

    fun addPrevItem(fragment: Fragment, tag: String) {
        fragmentPagerArray.add(0, fragment)
        fragmentTag.add(0, tag)
        notifyDataSetChanged()
    }

    fun addItem(fragment: Fragment, tag: String) {
        fragmentPagerArray.add(fragment)
        fragmentTag.add(tag)
        notifyDataSetChanged()
    }
}